src/dft/backends/portfft/commit.cpp

```diff
@@ -0,0 +1,320 @@
+/*******************************************************************************
+* Copyright Codeplay Software Ltd
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include <array>
+#include <optional>
+
+#include <portfft/portfft.hpp>
+
+#include "oneapi/mkl/exceptions.hpp"
+
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "oneapi/mkl/dft/detail/descriptor_impl.hpp"
+#include "oneapi/mkl/dft/detail/portfft/onemkl_dft_portfft.hpp"
+#include "oneapi/mkl/dft/types.hpp"
+
+#include "portfft_helper.hpp"
+
+// alias to avoid ambiguity
+namespace pfft = portfft;
+
+namespace oneapi::mkl::dft::portfft {
+namespace detail {
+
+/// Commit impl class specialization for cuFFT.
+template <dft::precision prec, dft::domain dom>
+class portfft_commit final : public dft::detail::commit_impl<prec, dom> {
+private:
+    using scalar_type = typename dft::detail::commit_impl<prec, dom>::scalar_type;
+    using fwd_type = typename dft::detail::commit_impl<prec, dom>::fwd_type;
+    using bwd_type = typename dft::detail::commit_impl<prec, dom>::bwd_type;
+    using descriptor_type = typename dft::detail::descriptor<prec, dom>;
+
+    static constexpr pfft::domain domain =
+        std::conditional_t<dom == dft::domain::REAL,
+                           std::integral_constant<pfft::domain, pfft::domain::REAL>,
+                           std::integral_constant<pfft::domain, pfft::domain::COMPLEX>>::value;
```

Comment:
Isn't this equivalent to the following?

```suggestion
    static constexpr pfft::domain domain = dom == dft::domain::REAL ? pfft::domain::REAL : pfft::domain::COMPLEX;
```

Reply:
Yes! I guess I overcomplicated because templates metaprogramming can't be that easy :sweat_smile: 
fixed in https://github.com/FMarno/oneMKL/pull/1/commits/fc720ead06549335a71310bb2dc44e976703488b